src/App.tsx

```diff
@@ -26,70 +29,127 @@ type ReorderOptions = {
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  hiddenReset: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: true,
+    sortType: SortType.NONE,
+    hiddenReset: false,
+  };
+
+  SortAlphabetically = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+      hiddenReset: true,
+    });
+  };
+
+  SortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      hiddenReset: true,
+    });
+  };
+
+  reset = () => {
+    this.setState(state => ({
+      isReversed: true,
+      sortType: SortType.NONE,
+      hiddenReset: !state.hiddenReset,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      hiddenReset: true,
+    }));
+  };
+
+  render() {
+    const {
+      isReversed,
+      sortType,
+      hiddenReset,
+    } = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': sortType !== SortType.ALPHABET })}
+            onClick={this.SortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success',
+              { 'is-light': sortType !== SortType.LENGTH })}
+            onClick={this.SortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning',
+              { 'is-light': isReversed })}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {hiddenReset && (
```

Comment:
you can also check if the list is reversed (isReversed) and sort type not equal NONE to not create extra variable `hiddenReset`